src/docs/sphinx/tutorial_cpp_basics.rst

```diff
@@ -159,7 +163,65 @@ You can check if a Schema is compatible with another Schema using the **Schema::
  - **If the calling Schema describes a leaf data type**: The calling Schema's and test Schema's **dtype().id()** and **dtype().element_bytes()** must match, and the calling Schema **dtype().number_of_elements()** must be greater than or equal than the test Schema's.
 
 
+Node References
+--------------------------------
+
+For most uses cases in C++, Node references are the best solution to build and
+manipulate trees. They allow you to avoid expensive copies and pass around
+sub-trees without worrying about valid pointers.
+
 
+.. literalinclude:: ../../tests/docs/t_conduit_docs_tutorial_basics.cpp
+   :start-after: BEGIN_EXAMPLE("basics_node_refs")
+   :end-before:  END_EXAMPLE("basics_node_refs")
+   :language: cpp
+   :dedent: 4
 
+.. literalinclude:: t_conduit_docs_tutorial_basics_out.txt
+   :start-after: BEGIN_EXAMPLE("basics_node_refs")
+   :end-before:  END_EXAMPLE("basics_node_refs")
 
 
+In C++ the Node assignment operator that takes a Node input is really an alias
+to set. That is, if follows set (deep copy) semantics.
+
+.. literalinclude:: ../../tests/docs/t_conduit_docs_tutorial_basics.cpp
+   :start-after: BEGIN_EXAMPLE("basics_node_refs_bad")
+   :end-before:  END_EXAMPLE("basics_node_refs_bad")
+   :language: cpp
+   :dedent: 4
+
+.. literalinclude:: t_conduit_docs_tutorial_basics_out.txt
+   :start-after: BEGIN_EXAMPLE("basics_node_refs_bad")
+   :end-before:  END_EXAMPLE("basics_node_refs_bad")
+
+
+const Nodes
+--------------------------------
+
+If you aren't careful, the ability to easily create dynamic trees can
+also undermine your process to consume them. 
+For example, asking for a expected but non-existent path will return
+a reference to an empty Node. Surprise!
+
+Methods like `fetch_existing` allow you to be more explicit
+when asking for expected data. In C++, ``const Node`` references are also common
+way to process trees in an read-only fashion. ``const`` methods will not modify
+the tree structure, so if you ask for a non-existent path, you will receive
+an error instead of reference to an empty Node.
```

Comment:
I get a lot of questions about this, so I'm very pleased to see this documentation; great addition!